engine/src/main/java/com/google/android/fhir/sync/Sync.kt

```diff
@@ -64,40 +62,43 @@ object Sync {
    * Starts a one time sync based on [FhirSyncWorker]. In case of a failure, [RetryConfiguration]
    * will guide the retry mechanism. Caller can set [retryConfiguration] to [null] to stop retry.
    */
-  inline fun <reified W : FhirSyncWorker> oneTimeSync(
+  inline fun <W : FhirSyncWorker> oneTimeSync(
     context: Context,
+    clazz: Class<W>,
     retryConfiguration: RetryConfiguration? = defaultRetryConfiguration
   ) {
     WorkManager.getInstance(context)
       .enqueueUniqueWork(
         SyncWorkType.DOWNLOAD.workerName,
         ExistingWorkPolicy.KEEP,
-        createOneTimeWorkRequest<W>(retryConfiguration)
+        createOneTimeWorkRequest(retryConfiguration, clazz)
       )
   }
   /**
    * Starts a periodic sync based on [FhirSyncWorker]. It takes [PeriodicSyncConfiguration] to
    * determine the sync frequency and [RetryConfiguration] to guide the retry mechanism. Caller can
    * set [retryConfiguration] to [null] to stop retry.
    */
-  inline fun <reified W : FhirSyncWorker> periodicSync(
+  inline fun <W : FhirSyncWorker> periodicSync(
     context: Context,
     periodicSyncConfiguration: PeriodicSyncConfiguration,
+    clazz: Class<W>
```

Comment:
```suggestion
  inline fun < reified W : FhirSyncWorker> periodicSync(
    context: Context,
    periodicSyncConfiguration: PeriodicSyncConfiguration)
```
Just like oneTimeSync, this api shouldn't change as well.